src/cpp/src/speculative_decoding/speculative_decoding_eagle_utils.cpp

```diff
@@ -0,0 +1,249 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "speculative_decoding_eagle_utils.hpp"
+
+#include <fstream>
+#include <nlohmann/json.hpp>
+
+#include "json_utils.hpp"
+#include "logger.hpp"
+
+#include "openvino/op/add.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/concat.hpp"
+
+namespace ov {
+namespace genai {
+namespace speculative_decoding {
+
+void ensure_num_assistant_tokens_is_set(ov::genai::GenerationConfig& config) {
+    // Only num_assistant_tokens is supported, not assistant_confidence_threshold
+    OPENVINO_ASSERT(
+        config.assistant_confidence_threshold == 0.f,
+        "Speculative Decoding only supports num_assistant_tokens, not assistant_confidence_threshold. Set it to 0.f.");
+    if (config.num_assistant_tokens == 0) {
+        config.num_assistant_tokens = DEFAULT_NUM_ASSISTANT_TOKENS;
+    }
+}
+
+Eagle3RTInfo
+extract_eagle_mode_from_config(ov::AnyMap& config, const std::filesystem::path& models_path) {
```

Comment:
Do we really need to change `config` here or it can be just `const ov::AnyMap& config`?